### ISSUE TITLE
Tensorflow 1.9 or 1.12 support

### ISSUE THREAD

**Issue by tsievers33** [ethz-asl/tensorflow_catkin#2]
I want to use this package for TensorFlow Object Detection, only this wrapper doesn't work for TensorFlow 1.9 or higher because of Dependency errors.

Maybe someone can help me! I will provide details (compile errors) if so...

**Comment by sarlinpe** [comment]
Hi, I currently don't have time to look at it but can give some pointers on how to proceed. 

First, change the commit tag in the [CMakeLists.txt](https://github.com/ethz-asl/tensorflow_catkin/blob/master/CMakeLists.txt#L27) to the one corresponding to TF 1.9.  If you're lucky, Catkin will compile it successfully. If it fails, it's either because the code patched for TF 1.8 (see [`patches/`](https://github.com/ethz-asl/tensorflow_catkin/tree/master/patches)) has changed in TF 1.9, and/or because some more patching is needed. In both cases, you'll have to look at the errors, do some patching, compile, and repeat. To make the process faster, you might want to instead directly [compile with CMake](https://github.com/tensorflow/tensorflow/tree/v1.9.0/tensorflow/contrib/cmake) and patch manually.

As the CMake build has been deprecated since TF 1.12, it might become much more difficult to make that work for TF >=1.12. I currently do not have any solution to this and do not plan to try to fix it in the future.

**Comment by tsievers33** [comment]
Hi, Thanks for your support, when I found a way to get Tensorflow 1.9 working, I will let it know!

**Comment by Karsten1987** [comment]
Hey, is there any update on this? I am also very interested to see this push forward.

**Comment by tsievers33** [comment]
Yes, I did make it work with TF 1.9, not via tensorflow_catkin but via Bazel buiild. Only TF 1.9, TF 1.12 did not succeed!

This one:
https://github.com/tradr-project/tensorflow_ros_cpp TF 1.9 is working!

**Comment by tsievers33** [comment]
Here are our ROS detection C++ files:
https://gitlab.twizzel.net/ros/object-detection 
For us it was a school project, maybe I could look for how we did it, but that will be tomorrow.

@Skydes look at https://gitlab.twizzel.net/ros/object-detection/blob/master/CMakeLists.txt
